### PR TITLE
Add Support for BPF arch

### PR DESCRIPTION
**Pull request by deepaksirone**
Hello,

Thanks for maintaining this project. This PR adds support for BPF in capstone-rs. I regenerated the bindings in capstone-sys using a newer version of bindgen in Cargo.toml since it did not work with clang-16. I am yet to write some tests for the implementation. Please let me know if I am on the right track with this or if I am missing something! :-)

### CODE REVIEW

**Comment by tmfink** [comment]
Thank you for the contribution! I'll take a look at your changes.

File: capstone-sys/pre_generated/capstone_archs_impl.rs
Comment (tmfink) [inline capstone-sys/pre_generated/capstone_archs_impl.rs:1]: Please read through the directions in `capstone-sys/pre_generated/update-bindings.md`; it looks like this file is unformatted

**Review by tmfink — CHANGES_REQUESTED** [review]
Overall the changes look good, but please also add tests in `capstone-rs/src/test.rs`.
You can find tests by building/running `capstone-sys/capstone/tests/test_bpf.c`.

**Comment by tmfink** [comment]
Aside from my  comments, I may want to wait to merge this until #143 is merged since that's a big change and there would be conflicts.

@Lancern do you have any opinions?

**Comment by Lancern** [comment]
> Aside from my comments, I may want to wait to merge this until #143 is merged since that's a big change and there would be conflicts.
> 
> @Lancern do you have any opinions?

The BPF arch added in this PR looks good and it should be easy to merge it with my PR. If you decide that this PR lands after my PR, I'll try finish my PR quickly.

**Comment by tmfink** [comment]
> The BPF arch added in this PR looks good and it should be easy to merge it with my PR. If you decide that this PR lands after my PR, I'll try finish my PR quickly.

@Lancern If you're fine handling any conflicts, then I'll just "merge" whichever PR is ready first.

Also, please don't literally `git merge`. Instead, please maintain linear history with something like `git rebase -i`.

**Comment by deepaksirone** [comment]
Hi @tmfink, I have added some tests and fixed the formatting of the generated files.

File: capstone-rs/src/test.rs [hunk 1]
```diff
@@ -3244,3 +3255,99 @@ fn test_owned_insn() {
...
+
+        Err(e) => {
+            eprintln!("{:?}", e);
+            assert!(false);
+        }
+    }
+
+}
+
+fn test_ebpf() {
```
Comment (tmfink) [inline capstone-rs/src/test.rs:3314]: Also add a test function `test_arch_bpf_detail()` that tests the details for each instruction. These `test_arch_*_detail()` tests catch a lot of bugs and changes in behavior when we upgrade the C library. For an example, see `test_arch_x86_detail()`:
   https://github.com/capstone-rust/capstone-rs/blob/7c9a51f6eb81709cde1de7f7498a0e16f29bd852/capstone-rs/src/test.rs#L2681

**Review by tmfink — CHANGES_REQUESTED** [review]
After making the changes, please run `cargo fmt` to format the entire repo.

**Comment by chengshuyi** [comment]
Hi, are there any plans to merge this PR? I look forward to using this feature. Thanks.

**Comment by deepaksirone** [comment]
Hi @tmfink and @chengshuyi I have added the tests requested. Please let me know if there are any more changes needed

File: capstone-rs/src/arch/mod.rs [hunk 2]
```diff
@@ -534,6 +544,13 @@ macro_rules! detail_arch_base {
                 /// Returns the XCore details, if any
                 => arch_name = xcore,
             ]
+            [
+                detail = BpfDetail,
+                insn_detail = BpfInsnDetail<'a>,
+                op = BpfOperand,
+                /// Returns the XCore details, if any
```
Comment (tmfink) [inline capstone-rs/src/arch/mod.rs:551]: copy/paste error: mention BPF instead of Xcore

File: capstone-rs/src/test.rs [hunk 3]
```diff
@@ -3244,3 +3255,185 @@ fn test_owned_insn() {
...
+                println!("{:4}operands: {}", "", ops.len());
+                for op in ops {
+                    eprintln!("{:8}{:?}", "", op);
+                }
+            }
+        }
+
+        Err(e) => {
+            eprintln!("{:?}", e);
+            assert!(false);
```
Comment (tmfink) [inline capstone-rs/src/test.rs:3358]: just call `panic!()` instead

File: capstone-rs/src/test.rs [hunk 4]
```diff
@@ -3244,3 +3255,185 @@ fn test_owned_insn() {
...
```
Comment (tmfink) [inline capstone-rs/src/test.rs:3312]: panic!() instead

**Review by tmfink — APPROVED** [review]
minor issues; I'll clean them up

**Comment by tmfink** [comment]
Thanks for the PR!
I'll do some minor clean-up myself

**Comment by tmfink** [comment]
Did some minor clean-up w/ 38bbcce944e42a44638f99489b669f4516cd7001

**Pull request merged**